src/svm/interface/svm.c

```diff
@@ -2067,6 +2067,55 @@ PetscErrorCode SVMTest(SVM svm)
   PetscFunctionReturnI(0);
 }
 
+#undef __FUNCT__
+#define __FUNCT__ "SVMConvergedSetUp"
+/*@
+
+@*/
+PetscErrorCode SVMConvergedSetUp(SVM svm)
+{
+
+  PetscFunctionBegin;
+  PetscValidHeaderSpecific(svm,SVM_CLASSID,1);
+  // call specific implementation of setting convergence test
+  if (svm->ops->convergedsetup) PetscCall(svm->ops->convergedsetup(svm));
+  PetscFunctionReturn(0);
+}
+
+#undef __FUNCT__
+#define __FUNCT__ "SVMDefaultConvergedCreate"
+/*@
+
+@*/
+PetscErrorCode SVMDefaultConvergedCreate(void **ctx, SVM svm)
+{
+  SVMConvergedCtx *cctx;
+
+  PetscFunctionBegin;
+  PetscValidHeaderSpecific(svm,SVM_CLASSID,1);
+
+  PetscCall(PetscNew(&cctx));
+  cctx->svm = svm;
+  PetscCall(PetscObjectReference((PetscObject) svm));
```

Comment:
Looking at SVMDestroy - this leaks svm object. Since refct = 2 
https://github.com/permon/permonsvm/blob/6d2386c2c6549664667c271688ca0c0bc05a814d/src/svm/interface/svm.c#L161
SVMDestroy() just decreases refct and sets svm=NULL.

See `DMDestroy` for handling cyclic references.